codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -780,53 +805,71 @@ private LastHttpContent readTrailingHeaders(ByteBuf buffer) {
     protected abstract HttpMessage createMessage(String[] initialLine) throws Exception;
     protected abstract HttpMessage createInvalidMessage();
 
-    private static int getChunkSize(String hex) {
-        hex = hex.trim();
-        for (int i = 0; i < hex.length(); i ++) {
-            char c = hex.charAt(i);
-            if (c == ';' || Character.isWhitespace(c) || Character.isISOControl(c)) {
-                hex = hex.substring(0, i);
-                break;
+    private static int getChunkSize(byte[] hex, int length) {
+        // hex shouldn't contain both ISO chars nor spaces
```

Comment:
hex should contain neither ISO chars nor spaces

Reply:
double negation - ooops, good point